pkg/knowledge_base2/graph.go

```diff
@@ -111,11 +112,13 @@ func firstFunctional(
 func allDeps(
 	ids *[]construct.ResourceId,
 ) graph_addons.WalkGraphFunc[construct.ResourceId] {
+	resourceSet := set.Set[construct.ResourceId]{}
```

Comment:
Aren't the other functions/layers supposed to have this same behaviour? If so, might as well just change the `*[]construct.ResourceId` to just a `set.Set[construct.ResourceId]` which would simplify the code here a bit.

Reply:
sure i can update to this